sql/sql_profile.cc

```diff
@@ -162,6 +163,63 @@ int make_profile_table_for_show(THD *thd, ST_SCHEMA_TABLE *schema_table) {
   }
   return 0;
 }
+typedef int (*je_mallctl_func)(const char *name, void *oldp, size_t *oldlenp,
+                               void *newp, size_t newlen);
+
+bool opt_jemalloc_profiling_enabled = false;
+bool opt_jemalloc_detected = false;
+static je_mallctl_func mallctl_p = NULL;
+static unsigned jemalloc_initialized = 0;
+static unsigned jemalloc_profile_counter = 0;
+
+int jemalloc_mallctl(const char *name, void *oldp, size_t *oldlenp, void *newp,
+                     size_t newlen) {
+  if (!mallctl_p && !jemalloc_initialized) {
+    mallctl_p = (je_mallctl_func)dlsym(RTLD_DEFAULT, "mallctl");
+    jemalloc_initialized = 1;
+  }
+
+  if (mallctl_p) return mallctl_p(name, oldp, oldlenp, newp, newlen);
+
+  return 1;
+}
+
+int jemalloc_profiling_dump() {
```

Comment:
Good points. I have solved points 1 and 2. About 3, I'd rather keep the return `int` although it is not checked. It doesn't hurt having that information available in case is needed.